data_tools/firebase_client_wrapper.py

```diff
@@ -138,44 +232,71 @@ def set_segment_messages_content_batch(dataset_id, messages, segment_index=None,
         batch_counter += 1
         if batch_counter >= batch_size:
             batch.commit()
-            print ("Batch of {} messages committed, progress: {} / {}".format(batch_counter, i, total_messages_count))
+            print("Batch of {} messages committed, progress: {} / {}".format(batch_counter, i, total_messages_count))
             batch_counter = 0
             batch = client.batch()
 
     if batch_counter > 0:
         batch.commit()
-        print ("Final batch of {} messages committed".format(batch_counter))
+        print("Final batch of {} messages committed".format(batch_counter))
+
+    print("Written {} messages".format(i))
 
-    print ("Written {} messages".format(i))
 
 def set_dataset_messages_content_batch(dataset_id, messages, batch_size=500):
     next_batch_to_write = []
     latest_segment_index = get_segment_count(dataset_id)
-    latest_segment_size = len(get_segment_messages(dataset_id, latest_segment_index))
+    latest_segment_size = len(get_segment_messages(id_for_segment(dataset_id, latest_segment_index)))
     for message in messages:
         if latest_segment_size >= MAX_SEGMENT_SIZE:
             set_segment_messages_content_batch(
                 dataset_id, next_batch_to_write, segment_index=get_segment_count(dataset_id), batch_size=batch_size)
             next_batch_to_write = []
-            create_next_dataset_segment(dataset_id)
+            create_next_segment(dataset_id)
             latest_segment_size = 0
         next_batch_to_write.append(message)
         latest_segment_size += 1
 
     set_segment_messages_content_batch(
         dataset_id, next_batch_to_write, segment_index=get_segment_count(dataset_id), batch_size=batch_size)
 
+
+def delete_segment(segment_id):
+    # Delete code schemes
+    segment_code_schemes_path = f"datasets/{segment_id}/code_schemes"
+    print(f"Deleting {segment_code_schemes_path}...")
+    _delete_collection(client.collection(segment_code_schemes_path), 10)
```

Comment:
This seems a pretty small batch size? I don't recall if delete batches have to be smaller?

Reply:
I believe they can be the max size, I was just following your lead of 10 at a time.

But if you're happy with an increase I will bump to the maximum